src/Nethermind/Nethermind.Optimism/CL/OptimismCL.cs

```diff
@@ -6,50 +6,53 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Nethermind.Core;
-using Nethermind.Core.Specs;
-using Nethermind.Crypto;
 using Nethermind.Logging;
 using Nethermind.Optimism.CL.Decoding;
-using Nethermind.Optimism.CL.Derivation;
 using Nethermind.Optimism.CL.L1Bridge;
 using Nethermind.Optimism.CL.P2P;
-using Nethermind.Optimism.Rpc;
-using Nethermind.Serialization.Json;
 
 namespace Nethermind.Optimism.CL;
 
-public class OptimismCL : IDisposable
+public sealed class OptimismCL : IDisposable
 {
-    private readonly DecodingPipeline _decodingPipeline;
+    private readonly IDecodingPipeline _decodingPipeline;
     private readonly IL1Bridge _l1Bridge;
-    private readonly L1ConfigValidator _l1ConfigValidator;
+    private readonly IL1ConfigValidator _l1ConfigValidator;
+    private readonly IL2Api _l2Api;
     private readonly IExecutionEngineManager _executionEngineManager;
+    private readonly ITimestamper _timestamper;
+
+    private readonly CLChainSpecEngineParameters _engineParameters;
+    private readonly ulong _l2BlockTime;
+    private readonly ulong _l2GenesisTimestamp;
+
     private readonly Driver _driver;
-    private readonly IL2Api _l2Api;
     private readonly OptimismCLP2P _p2p;
-    private readonly ITimestamper _timestamper;
+
     private readonly ILogger _logger;
-    private readonly CLChainSpecEngineParameters _engineParameters;
     private readonly CancellationTokenSource _cancellationTokenSource = new();
 
     public OptimismCL(
```

Comment:
Constructor had to be heavily modified to inject all dependencies instead of constructing it itself since they're also required by the RPC modules

Reply:
The validation case shows and this one show that the more we push towards DI the less manual management and duplication of instantiation is required.